symfony/client/src/orangehrmAdminPlugin/pages/qualificationSkill/EditQualificationSkill.vue

```diff
@@ -0,0 +1,164 @@
+      <oxd-text tag="h6">Edit Skill</oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form :loading="isLoading" @submitValid="onSave">
+        <oxd-form-row>
+          <oxd-input-field
+            label="Skill Name"
+            v-model="skill.name"
+            :rules="rules.name"
+            required
+          />
+        </oxd-form-row>
+
+        <oxd-form-row>
+          <oxd-input-field
+            type="textarea"
+            label="Skill Description"
+            placeholder="Type description here"
+            v-model="skill.description"
+            :rules="rules.description"
+          />
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <oxd-button type="button" displayType="ghost" label="Cancel" @click="onCancel" />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@orangehrm/core/util/services/api.service';
+
+const skillModel = {
+  id: '',
+  name: '',
+  description: '',
+};
+
+export default {
+  props: {
+    qualificationSkillId: {
+      type: Number,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      skill: {...skillModel},
+      rules: {
+        name: [],
+        description: [
+          v =>
+            (v && v.length < 400) ||
+            v === '' ||
+            'Should be less than 400 characters',
+        ],
+      },
+    };
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/admin/skills',
+    );
+    return {
+      http,
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .update(this.qualificationSkillId, {
+          name: this.skill.name,
+          description: this.skill.description,
+        })
+        .then(() => {
+          return this.$toast.success({
+            title: 'Success',
+            message: 'Qualification Skills updated successfully!',
+          });
+        })
+        .then(() => {
+          this.onCancel();
+          this.isLoading = false;
+        });
+    },
+    onCancel() {
+      navigate('/admin/viewSkills');
+    },
+  },
+  created() {
+    this.isLoading = true;
+    this.http
+      .get(this.qualificationSkillId)
+      .then(response => {
+        const {data} = response.data;
+        this.skill.id = data.id;
+        this.skill.name = data.name;
+        this.skill.description = data.description;
+
+        // Fetch list data for unique test
+        return this.http.getAll();
+      })
+      .then(response => {
+        const {data} = response.data;
+        this.rules.name.push(v => {
+          return (!!v && v.trim() !== '') || 'Required';
+        });
+        this.rules.name.push(v => {
+          return (v && v.length <= 100) || 'Should be less than 50 characters';
```

Comment:
Should be less than 100 characters

Reply:
Corrected this as "return (v && v.length <= 50) || 'Should be less than 50 characters';"
Since, as Chamara developed when saving, it allows only 50 characters.